remote_attestation/rust/src/message.rs

```diff
@@ -235,3 +270,62 @@ impl Deserializable for ClientIdentity {
         }
     }
 }
+
+impl EncryptedData {
+    pub fn new(nonce: &[u8; NONCE_LENGTH], data: &[u8]) -> Self {
+        Self {
+            header: ENCRYPTED_DATA_HEADER,
+            nonce: *nonce,
+            data: data.to_vec(),
+        }
+    }
+}
+
+impl Serializable for EncryptedData {
+    fn serialize(&self) -> anyhow::Result<Vec<u8>> {
+        bincode::serialize(&self).context("Couldn't serialize encrypted data message")
+    }
+}
+
+impl Deserializable for EncryptedData {
+    fn deserialize(bytes: &[u8]) -> anyhow::Result<Self> {
+        let message: Self =
+            bincode::deserialize(bytes).context("Couldn't deserialize encrypted data message")?;
+        if message.header == ENCRYPTED_DATA_HEADER {
+            Ok(message)
+        } else {
+            Err(anyhow!("Incorrect encrypted data message header"))
+        }
+    }
+}
+
+/// Deserializes an attestation message from a serialized [`input`] and wraps in a
+/// [`MessageWrapper`].
+pub fn deserialize(input: &[u8]) -> anyhow::Result<MessageWrapper> {
+    if input.is_empty() {
+        return Err(anyhow!("Empty input"));
+    }
+    match input[0] {
```

Comment:
This validates my point that there is really an outer structure for the frame and this should match on that outer structure.